opentelemetry-instrumentation/src/opentelemetry/instrumentation/metric.py

```diff
@@ -0,0 +1,85 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# type: ignore
+
+"""
+OpenTelemetry Instrumentation Metric mixin
+"""
+import enum
+from contextlib import contextmanager
+from time import time
+from typing import Dict, Optional
+
+from opentelemetry import metrics
+from opentelemetry.sdk.metrics import ValueRecorder
+
+
+class HTTPMetricType(enum.Enum):
+    CLIENT = 0
+    SERVER = 1
+    # TODO: Add both
```

Comment:
This enumerator is currently being used to add more accurate description, what is the value of supporting both?

Reply:
The enumerator is not only for specifying the description. There are actually different labels that are needed depending on whether the type is `client` or `server`. "Adding both" refers to libraries that might actually emit both types, in which we would then have to create separate metric instances for duration.